cmake-modules/download_test_data.cmake

```diff
@@ -0,0 +1,54 @@
+# this module is used and required by the nlohmann json lib unit tests.
+# It gets the test data
+
+find_package(Git)
+
+set(JSON_TEST_DATA_URL     https://github.com/nlohmann/json_test_data)
+set(JSON_TEST_DATA_VERSION 2.0.0)
+
+# target to download test data
+add_custom_target(download_test_data
+    COMMAND test -d json_test_data || ${GIT_EXECUTABLE} clone -c advice.detachedHead=false --branch v${JSON_TEST_DATA_VERSION} ${JSON_TEST_DATA_URL}.git --quiet --depth 1
+    COMMENT "Downloading test data from ${JSON_TEST_DATA_URL} (v${JSON_TEST_DATA_VERSION})"
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+)
```

Comment:
Add link to where you got this logic from. I am guessing:
https://github.com/nlohmann/json/blob/fd7a9f600712b2724463e9f7f703878ade676d6e/cmake/download_test_data.cmake#L12